contracts/test/MerkleGrantRoundPayout.test.ts

```diff
@@ -0,0 +1,264 @@
+// --- External imports ---
+import { artifacts, ethers, waffle } from 'hardhat';
+import { deployMockContract, MockContract } from 'ethereum-waffle';
+
+// --- Our imports ---
+import { MerkleGrantRoundPayout } from '../typechain/MerkleGrantRoundPayout';
+import { expect } from 'chai';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import { Artifact } from 'hardhat/types';
+import { balanceOf, setBalance, tokens } from './utils';
+import { BalanceTree } from '../../utils/src/merkle-distributor/balance-tree';
+import { BigNumber, Contract } from 'ethers';
+
+// --- Parse and define helpers ---
+const { deployContract } = waffle;
+const { isAddress } = ethers.utils;
+const randomAddress = () => ethers.Wallet.createRandom().address;
+
+type Claim = {
+  index: number;
+  payee: string;
+  amount: number;
+  merkleProof: string[];
+};
+
+const ZERO_BYTES32 = '0x0000000000000000000000000000000000000000000000000000000000000000';
+
+describe('MerkleGrantRoundPayout', function () {
+  let user: SignerWithAddress;
+  let mockToken: MockContract;
+  let payout: MerkleGrantRoundPayout;
+
+  before(async () => {
+    [user] = await ethers.getSigners();
+
+    // Deploy mock contracts
+    mockToken = await deployMockContract(user, ['function totalSupply() returns(uint256)']);
+    await mockToken.mock.totalSupply.returns('0');
+
+    // Deploy PayoutContract
+    const payoutArtifact: Artifact = await artifacts.readArtifact('MerkleGrantRoundPayout');
+    payout = <MerkleGrantRoundPayout>await deployContract(user, payoutArtifact, [tokens.dai.address, ZERO_BYTES32]);
+  });
+
+  describe('constructor', () => {
+    it('deploys properly', async function () {
+      // Verify deploy
+      expect(isAddress(payout.address), 'Failed to deploy MerkleGrantRoundPayout').to.be.true;
+
+      // Verify constructor parameters
+      expect(await payout.token()).to.equal(tokens.dai.address);
+      expect(await payout.merkleRoot()).to.equal(ZERO_BYTES32);
+    });
+  });
+
+  describe('check state variable', () => {
+    it('returns the token address', async () => {
+      expect(await payout.token()).to.eq(tokens.dai.address);
+    });
+    it('returns the zero merkle root', async () => {
+      expect(await payout.merkleRoot()).to.eq(ZERO_BYTES32);
+    });
+  });
+
+  describe('claim', () => {
+    let payout: Contract;
+    let tree: BalanceTree;
+    // user0
+    let user0: SignerWithAddress;
+    let proof0: string[];
+    let claim0Arg: Claim;
+    // user1
+    let user1: SignerWithAddress;
+    let proof1: string[];
+    let claim1Arg: Claim;
+
+    beforeEach(async () => {
+      const signers = await ethers.getSigners();
+      user0 = signers[0];
+      user1 = signers[1];
```

Comment:
Nit, but can be a bit more concise here with
```suggestion
      const [user0, user1] = await ethers.getSigners();
```